src/System.Diagnostics.Process/tests/ProcessStreamReadTests.cs

```diff
@@ -228,6 +228,23 @@ public void TestManyOutputLines()
             Assert.Equal(ExpectedLineCount + 1, totalLinesReceived);
         }
 
+        [Fact]
+        public void TestClosingStreamsDoesNotThrow()
+        {
+            Process p = CreateProcessPortable(RemotelyInvokable.WriteLinesSlowlyToOutputAndError);
+            p.StartInfo.RedirectStandardOutput = true;
+            p.StartInfo.RedirectStandardError = true;
+
+            p.OutputDataReceived += (s, e) => {};
+            p.ErrorDataReceived += (s, e) => {};
+
+            p.Start();
+            p.BeginOutputReadLine();
+            p.BeginErrorReadLine();
+
```

Comment:
Isn't this tested by a couple of the other tests that use the async mode? I think the fault surface introduced in this change is what happens if the child continues to write to stdout/stderr after they're closed, which I think is properly tested in `TestClosingStreamsAsyncDoesNotThrow` and `TestClosingStreamsUndefinedDoesNotThrow`

Reply:
They just don't call `Close` afterwards.
However, it would be more testing of the cancellation call discussed above. That is more part of general streams, nothing special here, so it's fine as it is now.